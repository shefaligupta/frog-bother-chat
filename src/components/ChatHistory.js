import React, {useEffect, useRef} from 'react';

/**
 * ChatHistory Class
 *
 * @param {Array} chatHistory
 * @returns {*}
 * @constructor
 */
const ChatHistory = ({chatHistory}) => {
  const messagesEndRef = useRef(null);

  /**
   * scrolls to bottom of the chat history.
   */
  const scrollToBottom = () => {
    if (messagesEndRef && messagesEndRef.current) {
      messagesEndRef.current.scrollIntoView({behavior: 'smooth'});
    }
  };
  useEffect(scrollToBottom, [chatHistory]);

  return (
      <div className="chat-history">
        {
          chatHistory
          && chatHistory.length > 0
          && chatHistory.map((chat) => (
              <div key={`message-${chat.timePeriod}`} className="message">
                <div key={`timestamp-${chat.timePeriod}`} className="message-time">{chat.timePeriod}</div>
                <div key={`mesg-${chat.timePeriod}`}>{chat.message}</div>
              </div>
          ))
        }
        <div ref={messagesEndRef}/>
      </div>
  );
};

export default ChatHistory;
