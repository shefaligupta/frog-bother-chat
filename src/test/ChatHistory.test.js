import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import ChatHistory from '../components/ChatHistory';


it('should render Henchmen list', () => {
  const tree = renderer.create(<ChatHistory/>).toJSON();
  expect(tree).toMatchSnapshot();
});
it('should render the message', () => {
  const chatHistory = [{
    message: 'This is a test message',
    timePeriod: 'Thu, 23 Apr 2020 01:36:37 GMT',
  }];

  const {container} = render(<ChatHistory {...chatHistory} />);
  const chatDiv = container.querySelector('[class="chat-history"]');
  expect(chatDiv.children).toHaveLength(1);
});
