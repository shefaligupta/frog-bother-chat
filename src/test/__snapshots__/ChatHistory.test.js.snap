// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Henchmen list 1`] = `
<div
  className="chat-history"
>
  <div />
</div>
`;
